### ISSUE TITLE
Ember Data Serializer should serialize API errors according to specification.

### ISSUE THREAD

**Issue by bummzack** [colymba/silverstripe-restfulapi#51]
The current API error payload doesn't match the ember [json-api specification](http://jsonapi.org/format/#errors). Here's how it currently looks:

    {message: "Token expired.", code: 3}

Here's how it should look:

    { errors: [{ title: "Token expired.", code: "3", status: 403 }] }

**Comment by colymba** [comment]
True. But the whole EmberDataSerializer is now far rom the json api spec. I have planned to deprecate the EmberDataSerializer in favour of a new JSONApiSerializer and try to follow the spec since it now is on v1...

**Comment by peavers** [comment]
Any update on the movement towards a new JSONApiSerializer? Would be really helpful at this stage and as you mention the spec has stabilized a fair amount.

**Comment by colymba** [comment]
@peavers yes, this needs addressing and I must say I haven't had much time... sorry...
I guess now is a good time to revive this and maybe have it in the next release that would be SS4 compatible....
